src/inc/msquic.h

```diff
@@ -241,6 +241,7 @@ typedef enum QUIC_SEND_FLAGS {
     QUIC_SEND_FLAG_DGRAM_PRIORITY           = 0x0008,   // Indicates the datagram is higher priority than others.
     QUIC_SEND_FLAG_DELAY_SEND               = 0x0010,   // Indicates the send should be delayed because more will be queued soon.
     QUIC_SEND_FLAG_CANCEL_ON_LOSS           = 0x0020,   // Indicates that a stream is to be cancelled when packet loss is detected.
+    QUIC_SEND_FLAG_DGRAM_CANCEL_ON_BLOCKED  = 0x0030,   // Indicates that a datagram should be dropped when it can't be sent immediately.
```

Comment:
I do wonder if a similar flag would be useful for streams (in the future), so should we remove the `DGRAM` from the name...

Reply:
Shall we change the function name / position to another file then such as "send.c", it would leave us with "QuicSendCancelOnBlocked" smth like that

Reply:
No. These are meant to be for the send requests, which is not the same thing as what send.c deals with. The location here is good.

Reply:
Well, what do you think of "QuicDatagramCancelOnBlocked" then, but we'll have to change that in the future if you want to include the same for Streams ?

Reply:
Sounds good; maybe just `QuicDatagramCancelBlocked`. In the future, we can add a `QuicConnectionCancelBlockedStreams`.